### PR TITLE
Implement VC 2.0 data model

### PR DESCRIPTION
**Pull request by kezike**
This PR adds/modifies fields for the VC 2.0 data model.

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
@kezike why the addition of `;` everywhere? The original library is going for a more Standard.js no-semicolon style.

File: src/VerifiableCredential.ts [hunk 1]
```diff
@@ -38,82 +39,124 @@ export interface VCDIVerifiableCredential extends LinkedDataObject {
...
   // https://w3c.github.io/vc-data-model/#status
-  credentialStatus?: CredentialStatus
+  credentialStatus?: CredentialStatus;
 
   // https://w3c.github.io/vc-data-model/#data-schemas
-  credentialSchema?: CredentialSchema
+  credentialSchema?: CredentialSchema | CredentialSchema[];
+
+  // https://w3c.github.io/vc-data-model/#integrity-of-related-resources
+  relatedResource?: RelatedResource[];
```
Comment (dmitrizagidulin) [inline src/VerifiableCredential.ts]:
```suggestion
  relatedResource?: RelatedResource | RelatedResource[];
```

File: src/VerifiableCredential.ts [hunk 2]
```diff
@@ -38,82 +39,124 @@ export interface VCDIVerifiableCredential extends LinkedDataObject {
...
+  [x: string]: any;
 }
 
 // https://w3c.github.io/vc-data-model/#status
 export interface CredentialStatus extends LinkedDataObject {
   // id and type are required for `credentialStatus` by the VC spec
-  id: string
-  type: Type
-  [x: string]: any
+  id?: string;
```
Comment (dmitrizagidulin) [inline src/VerifiableCredential.ts:106]: good catch!

**Review by dmitrizagidulin — APPROVED** [review]
👍 nice job on this. (I'd slightly prefer to go the non-semicolon route, tho.)

File: src/VerifiableCredential.ts (hunk 1 above)
Comment (kezike) [reply src/VerifiableCredential.ts]: I wanted to put this too, but I couldn't find an example of a single object anywhere. Plus, I found `array of objects` in the spec:

> To validate that a resource referenced by a [verifiable credential](https://w3c.github.io/vc-data-model/#dfn-verifiable-credential) is the same at verification time as it is at issuing time, an implementer MAY include a property named relatedResource that stores an array of objects that describe additional integrity metadata about each resource referenced by the [verifiable credential](https://w3c.github.io/vc-data-model/#dfn-verifiable-credential). If relatedResource is present, there MUST be an object in the array for each remote resource for each context used in the verifiable credential.

If you think it's useful to have anyway, I can add it.

Comment (dmitrizagidulin) [reply src/VerifiableCredential.ts]: Yeah, basically all objects in JSON-LD are either single or array (this technically even applies to issuer). The fact that there's no example of it for related resource is incidental.

**Comment by dmitrizagidulin** [comment]
Merging, thanks!

**Pull request merged**